cmd/crates/soroban-spec-typescript/fixtures/test_custom_types/src/invoke.ts

```diff
@@ -41,22 +31,40 @@ async function getAccount(): Promise<Account | null> {
 
 export class NotImplementedError extends Error { }
 
+type Simulation = SorobanClient.SorobanRpc.SimulateTransactionResponse
+type SendTx = SorobanClient.SorobanRpc.SendTransactionResponse
+type GetTx = SorobanClient.SorobanRpc.GetTransactionResponse
+
+// defined this way so typeahead shows full union, not named alias
+let someRpcResponse: Simulation | SendTx | GetTx
+type SomeRpcResponse = typeof someRpcResponse
+
+type InvokeArgs<R extends ResponseTypes, T = string> = Options<R> & {
+  method: string,
+  args?: any[],
+  parseResultXdr?: (xdr: string) => T,
+}
+
 /**
  * Invoke a method on the test_custom_types contract.
  *
  * Uses Freighter to determine the current user and if necessary sign the transaction.
  *
- * @param {string} obj.method - The method to invoke.
- * @param {any[]} obj.args - The arguments to pass to the method.
- * @param {boolean} obj.signAndSend - Whether to sign and send the transaction, or just simulate it. Unless the method requires authentication.
- * @param {number} obj.fee - The fee to pay for the transaction.
- * @returns The transaction response, or the simulation result if signing isn't required.
+ * @returns T, by default, the parsed XDR from either the simulation or the full transaction. If `simulateOnly` or `fullRpcResponse` are true, returns either the full simulation or the result of sending/getting the transaction to/from the ledger.
```

Comment:
nit: should be `@returns {T}  by default, <etc...>`, right?

Reply:
Fair point. Never knew you could reference a generic in a type doc.